src/usr/local/www/firewall_rules_edit.php

```diff
@@ -1881,20 +1881,12 @@ function typesel_change() {
 	}
 
 	function proto_change() {
-		if ($('#proto').find(":selected").index() < 3) {
-			portsenabled = 1;
-			hideClass('tcpflags', false);
-		} else {
-			portsenabled = 0;
-			hideClass('tcpflags', true);
-		}
+		var is_tcpudp = (jQuery.inArray($('#proto :selected').val(), ['tcp','udp', 'tcp/udp']) != -1);
+		portsenabled = (is_tcpudp ? 1 : 0);
```

Comment:
New line 1885 - I've left this value as it was (ie explicit 1 or 0) for ease of review. It's very likely that this line should just read "portsenabled = is_tcpudp;". if so, please make this change.